datus/agent/node/generate_sql_node.py

```
@@ -200,3 +233,73 @@ async def _generate_sql_stream(
+        if sql_query_dict and isinstance(sql_query_dict, dict):
+            return GenerateSQLResult(
+                success=True,
+                error=None,
+                sql_query=sql_query_dict.get("sql", ""),
+                tables=sql_query_dict.get("tables", []),
+                explanation=sql_query_dict.get("explanation"),
+            )
+        else:
+            return GenerateSQLResult(success=False, error="sql generation failed, no result", sql_query=sql_query)
+    except json.JSONDecodeError as e:
+        logger.error(f"SQL json decode failed: {str(e)} SQL: {sql_query}")
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)